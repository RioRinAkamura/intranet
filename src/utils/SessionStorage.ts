export class SessionStorage {
  // store key/value pair
  public async set(key: string, value: string): Promise<void> {
    localStorage.setItem(key, value);
  }

  // get value of
  public async get(key: string): Promise<string> {
    const value = localStorage.getItem(key);
    return value || '';
  }

  // delete key
  public async remove(key: string): Promise<void> {
    localStorage.removeItem(key);
  }
}
