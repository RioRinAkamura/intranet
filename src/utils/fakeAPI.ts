// First we need to import axios.js
import axios from 'axios';
// Next we make an 'instance' of it
const instance = axios.create({
  // .. where we make our configurations
  baseURL: 'https://api.boilerplate.dev.hdwebsoft.co/v1',
});

// Also add/ configure interceptors && all the other cool stuff

instance.interceptors.request.use(
  config => {
    if (!config.headers.Authorization) {
      const auth = localStorage.getItem('AUTH_SESSION_KEY');
      if (auth) {
        const token = JSON.parse(auth).access;
        if (token) {
          config.headers.Authorization = `Bearer ${token}`;
        }
      }
    }

    return config;
  },
  error => {
    console.log(error);
    return Promise.reject(error);
  },
);

instance.interceptors.response.use(
  response => {
    // Edit response config
    return response.data;
  },
  error => {
    console.log(error);
    return Promise.reject(error);
  },
);

export default instance;
