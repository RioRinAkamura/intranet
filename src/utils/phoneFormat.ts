export function phoneFormat(phone: string): string {
  if (phone.length === 10) {
    return phone.replace(/(\d{3})(\d{3})(\d{4})/, `$1 $2 $3`);
  } else if (phone.length === 11) {
    return phone.replace(/(\d{4})(\d{3})(\d{4})/, `$1 $2 $3`);
  } else {
    // This is temporary, if there are other phone format, add more
    return phone;
  }
}
