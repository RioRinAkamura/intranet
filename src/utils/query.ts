import querystring from 'query-string';
export function getQueryParam<T>(): T {
  if (typeof window === 'undefined') return {} as T;
  return querystring.parse(window.location.search) as any;
}