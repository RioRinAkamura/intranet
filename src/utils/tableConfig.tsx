import * as React from 'react';
import { has } from 'lodash';
import {
  Button,
  Checkbox,
  CheckboxOptionType,
  Col,
  Input,
  Row,
  Space,
  Select,
  Modal,
} from 'antd';
import { SearchOutlined } from '@ant-design/icons';
import { useTranslation } from 'react-i18next';
import Highlighter from 'react-highlight-words';
import { TagsInput } from 'app/components/Tags';
import { TableStateProps } from 'app/pages/UserPage/UserListPage/useHandleDataTable';
import { MessageTranslate, TagType } from './types';
import styled from 'styled-components/macro';
import { FilterColumns } from 'app/pages/UserPage/UserListPage/slice/types';
import { useProjectDetail } from 'app/pages/ProjectPage/ProjectDetailPage/useProjectDetail';

interface useTableProps {
  getColumnSorterProps: (dataIndex: string, columnPriority: number) => {};
  getColumnSearchInputProps: (dataIndex: string[], filterIndex?: number) => {};
  getColumnSearchTagProps: (dataIndex: string, tags?: TagType[]) => {};
  getColumnSearchCheckboxProps: (
    dataIndex: string[],
    options: CheckboxOptionType[],
    filterIndex?: number,
  ) => {};
  getColumnSearchCheckboxFromToProps: (
    dataIndex: string[],
    options: CheckboxOptionType[],
    range: string,
    filterIndex?: number,
  ) => {};
  ConfirmModal: () => JSX.Element;
}

const { Option } = Select;

export const useTableConfig = (
  state: TableStateProps,
  messageTrans: MessageTranslate,
  setFilterText: (value: FilterColumns) => void,
): useTableProps => {
  const { t } = useTranslation();
  const { update } = useProjectDetail();
  const [selectedKeys, setSelectedKeys] = React.useState([]);
  const [visible, setVisible] = React.useState(false);
  const [formValue, setFormValue] = React.useState({});

  React.useLayoutEffect(() => {
    if (state.filterColumns) {
      setSelectedKeys(prev => ({ ...prev, ...state.filterColumns }));
    }
  }, [state.filterColumns]);

  const handleOkConfirmModal = async () => {
    try {
      await update(formValue);
      setVisible(false);
    } catch (e) {
      console.log(e);
    }
  };

  const handleCancelConfirmModal = () => {
    setVisible(false);
  };

  const ConfirmModal = () => (
    <Modal
      title="Confirm Modal"
      visible={visible}
      onOk={handleOkConfirmModal}
      onCancel={handleCancelConfirmModal}
      okText="Ok"
      cancelText="Cancel"
    >
      <p>Are you sure this project has been completed?</p>
    </Modal>
  );

  const getColumnSorterProps = (dataIndex: string, columnPriority: number) => {
    const ordering = {
      sorter: {
        multiple: columnPriority,
      },
    };
    if (state.params.ordering) {
      ordering['sortOrder'] = state.params.ordering.includes(dataIndex)
        ? state.params.ordering.includes('-' + dataIndex)
          ? ('descend' as 'descend')
          : ('ascend' as 'ascend')
        : null;
    } else if (state.params.ordering === '') {
      ordering['sortOrder'] = null;
    }
    return ordering;
  };

  const getColumnSearchInputProps = (
    dataIndex: string[],
    filterIndex?: number,
  ) => ({
    ellipsis: true,
    filterDropdown: ({ confirm }) => {
      return (
        <Wrapper>
          <Input
            placeholder={`${t(
              messageTrans.filterInputPlaceholder(),
            )} ${dataIndex}`}
            value={selectedKeys[dataIndex[filterIndex || 0]]}
            onChange={e => {
              e.persist();
              setSelectedKeys(prevState => ({
                ...prevState,
                [dataIndex[filterIndex || 0]]: e.target.value
                  ? e.target.value
                  : null,
              }));
            }}
            onPressEnter={() =>
              handleSearch(dataIndex[filterIndex || 0], confirm)
            }
            style={{ width: 188, marginBottom: 8, display: 'block' }}
          />
          <Space>
            <Button
              type="primary"
              onClick={() => handleSearch(dataIndex[filterIndex || 0], confirm)}
              icon={<SearchOutlined />}
              size="small"
              style={{ width: 90 }}
              loading={state.loading}
            >
              {t(messageTrans.filterSearchButton())}
            </Button>
            <Button
              onClick={() => handleReset(dataIndex[filterIndex || 0], confirm)}
              size="small"
              loading={state.loading}
              style={{ width: 90 }}
            >
              {t(messageTrans.filterResetButton())}
            </Button>
          </Space>
        </Wrapper>
      );
    },
    filterIcon: filtered => (
      <SearchOutlined style={{ color: filtered ? '#1890ff' : undefined }} />
    ),
    onFilter: (value, record) =>
      record[dataIndex[filterIndex || 0]]
        ? record[dataIndex[filterIndex || 0]]
            .toString()
            .toLowerCase()
            .includes(value.toLowerCase())
        : '',
    render: (text, record) => {
      let dataText = '';
      dataIndex.map(data => {
        if (record[data]) {
          dataText += record[data] + ' ';
        }
        return data;
      });
      return has(state.filterColumns, dataIndex[filterIndex || 0]) ||
        (state.params.search && state.params.search.length > 0) ? (
        <Highlighter
          highlightStyle={{ backgroundColor: '#ffc069', padding: 0 }}
          searchWords={[
            state.filterColumns![dataIndex[filterIndex || 0]],
            state.params.search &&
              state.params.search.length > 0 &&
              state.params.search,
          ]}
          autoEscape
          textToHighlight={text ? dataText.trim().toString() : ''}
        />
      ) : (
        dataText.trim()
      );
    },
  });

  const handleSearch = (dataIndex: string, confirm: () => void) => {
    setFilterText({ [dataIndex]: selectedKeys[dataIndex] });
    confirm();
  };

  const handleReset = (dataIndex: string, confirm: () => void) => {
    setSelectedKeys(prevState => ({
      ...prevState,
      [dataIndex]: undefined,
    }));
    setFilterText({ [dataIndex]: undefined });
    confirm();
  };

  const handleFromToSearch = (
    dataIndex: string,
    range: string,
    confirm: () => void,
  ) => {
    if (selectedKeys[dataIndex] === '1') {
      setFilterText({ to: range, from: undefined });
    } else if (selectedKeys[dataIndex] === '2') {
      setFilterText({ from: range, to: range });
    } else {
      setFilterText({ from: range, to: undefined });
    }
    confirm();
  };

  const handleFromToReset = (dataIndex: string, confirm: () => void) => {
    setSelectedKeys(prevState => ({
      ...prevState,
      [dataIndex]: undefined,
      from: undefined,
      to: undefined,
    }));
    setFilterText({ from: undefined, to: undefined });
    confirm();
  };

  const getColumnSearchTagProps = (dataIndex: string) => ({
    filterDropdown: ({ confirm }) => {
      return (
        <Wrapper>
          <TagsInput
            value={selectedKeys[dataIndex]}
            callback={e => {
              setSelectedKeys(prevState => ({
                ...prevState,
                [dataIndex]: e ? e : null,
              }));
            }}
          />
          <Space>
            <Button
              type="primary"
              onClick={() => handleSearch(dataIndex, confirm)}
              icon={<SearchOutlined />}
              size="small"
              style={{ width: 90 }}
              loading={state.loading}
            >
              {t(messageTrans.filterSearchButton())}
            </Button>
            <Button
              onClick={() => handleReset(dataIndex, confirm)}
              size="small"
              loading={state.loading}
              style={{ width: 90 }}
            >
              {t(messageTrans.filterResetButton())}
            </Button>
          </Space>
        </Wrapper>
      );
    },
    filterIcon: filtered => (
      <SearchOutlined style={{ color: filtered ? '#1890ff' : undefined }} />
    ),
    onFilter: (value, record) =>
      record[dataIndex]
        ? record[dataIndex]
            .toString()
            .toLowerCase()
            .includes(value.toLowerCase())
        : '',
  });

  const getColumnSearchCheckboxProps = (
    dataIndex: string[],
    options: CheckboxOptionType[],
    filterIndex?: number,
  ) => ({
    filterDropdown: ({ confirm }) => {
      return (
        <Wrapper>
          <WrapperCheckbox>
            <Checkbox.Group
              value={selectedKeys[dataIndex[filterIndex || 0]]}
              options={options}
              onChange={e => {
                setSelectedKeys(prevState => ({
                  ...prevState,
                  [dataIndex[filterIndex || 0]]: e ? e : null,
                }));
              }}
            />
          </WrapperCheckbox>
          <Row gutter={[8, 0]}>
            <Col>
              <Button
                type="primary"
                onClick={() =>
                  handleSearch(dataIndex[filterIndex || 0], confirm)
                }
                icon={<SearchOutlined />}
                size="small"
                style={{ width: 90 }}
                loading={state.loading}
              >
                {t(messageTrans.filterSearchButton())}
              </Button>
            </Col>
            <Col>
              <Button
                onClick={() =>
                  handleReset(dataIndex[filterIndex || 0], confirm)
                }
                size="small"
                loading={state.loading}
                style={{ width: 90 }}
              >
                {t(messageTrans.filterResetButton())}
              </Button>
            </Col>
          </Row>
        </Wrapper>
      );
    },
    onFilter: (value, record) =>
      record[dataIndex[filterIndex || 0]]
        ? record[dataIndex[filterIndex || 0]]
            .toString()
            .toLowerCase()
            .includes(value.toLowerCase())
        : '',
    render: (text, record) => {
      let dataText = '';
      dataIndex.map(data => {
        if (record[data]) {
          dataText += record[data] + ' ';
        }
        return data;
      });
      const handleValueChange = async value => {
        const defaultForm = {
          ...record,
        };
        defaultForm[dataIndex[0]] = value;

        // hard code
        if (dataIndex[0] === 'status' && value === '4') {
          setVisible(true);
          setFormValue({ ...defaultForm });
          return;
        }

        try {
          await update(defaultForm);
        } catch (e) {
          console.log(e);
        }
      };
      let defaultValue: string = '';
      const findOption = options.find(
        option => option.value === Number(dataText),
      );
      defaultValue = findOption
        ? findOption.label
          ? findOption.label.toString()
          : ''
        : '';
      return has(state.filterColumns, dataIndex[filterIndex || 0]) ||
        (state.params.search && state.params.search.length > 0) ? (
        <Select
          onChange={handleValueChange}
          defaultValue={defaultValue}
          style={{ width: '100%' }}
        >
          {options.map(option => {
            return (
              <Option value={`${option.value}`}>
                <Highlighter
                  highlightStyle={{ backgroundColor: '#ffc069', padding: 0 }}
                  searchWords={[
                    state.filterColumns![dataIndex[filterIndex || 0]]?.includes(
                      text,
                    ) &&
                      options.find(option => option.value === Number(text))
                        ?.label,
                    state.params.search &&
                      state.params.search.length > 0 &&
                      state.params.search,
                  ]}
                  autoEscape
                  textToHighlight={option.label}
                />
              </Option>
            );
          })}
        </Select>
      ) : (
        <Select
          onChange={handleValueChange}
          defaultValue={defaultValue}
          style={{ width: '100%' }}
        >
          {options.map(option => {
            return <Option value={`${option.value}`}>{option.label}</Option>;
          })}
        </Select>
        // options.find(option => option.value === Number(dataText))?.label
      );
    },
  });

  const getColumnSearchCheckboxFromToProps = (
    dataIndex: string[],
    options: CheckboxOptionType[],
    range: string,
    filterIndex?: number,
  ) => ({
    filterDropdown: ({ confirm }) => {
      return (
        <Wrapper>
          <WrapperCheckbox>
            <Checkbox.Group
              value={selectedKeys[dataIndex[filterIndex || 0]]}
              options={options}
              onChange={e => {
                setSelectedKeys(prevState => ({
                  ...prevState,
                  [dataIndex[filterIndex || 0]]: e.toString()
                    ? e.toString()
                    : null,
                }));
              }}
            />
          </WrapperCheckbox>
          <Row gutter={[8, 0]}>
            <Col>
              <Button
                type="primary"
                onClick={() =>
                  handleFromToSearch(
                    dataIndex[filterIndex || 0],
                    range,
                    confirm,
                  )
                }
                icon={<SearchOutlined />}
                size="small"
                style={{ width: 90 }}
                loading={state.loading}
              >
                {t(messageTrans.filterSearchButton())}
              </Button>
            </Col>
            <Col>
              <Button
                onClick={() =>
                  handleFromToReset(dataIndex[filterIndex || 0], confirm)
                }
                size="small"
                loading={state.loading}
                style={{ width: 90 }}
              >
                {t(messageTrans.filterResetButton())}
              </Button>
            </Col>
          </Row>
        </Wrapper>
      );
    },
    onFilter: (value, record) =>
      record[dataIndex[filterIndex || 0]]
        ? record[dataIndex[filterIndex || 0]]
            .toString()
            .toLowerCase()
            .includes(value.toLowerCase())
        : '',
  });

  return {
    getColumnSorterProps,
    getColumnSearchInputProps,
    getColumnSearchTagProps,
    getColumnSearchCheckboxProps,
    getColumnSearchCheckboxFromToProps,
    ConfirmModal,
  };
};

const Wrapper = styled.div`
  padding: 8px;
`;

const WrapperCheckbox = styled.div`
  .ant-checkbox-group {
    display: grid;
  }

  border-bottom: 1px solid #d5d4d5;
  padding-bottom: 7px;
  margin-bottom: 7px;
`;
