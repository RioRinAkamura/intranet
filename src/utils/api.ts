import { Api, createApiClient } from '@hdwebsoft/boilerplate-api-sdk';
import { SessionStorage } from './SessionStorage';

export const sessionStorage = new SessionStorage();
const config = {
  baseUrl: 'https://api.boilerplate.dev.hdwebsoft.co/v1',
  authSessionKey: 'AUTH_SESSION_KEY',
  session: sessionStorage,
  socketUrl: 'http://template-ws.dev.goldfishcode.com',
};

const client = createApiClient(config);
export const api = new Api(client);
