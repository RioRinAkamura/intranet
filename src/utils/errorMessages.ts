import { translations } from 'locales/translations';
import { _t } from 'utils/messages';

export interface ErrorType {
  code: number;
  message: string;
  status_code: number;
  summary: string;
}

export const errorMess = {
  400: () => _t(translations.Errors.error400, 'Bad Request'),
  401: () => _t(translations.Errors.error401, 'Unauthorized'),
  403: () => _t(translations.Errors.error403, 'Forbidden'),
  404: () => _t(translations.Errors.error404, 'Not Found'),
  500: () => _t(translations.Errors.error500, 'Internal Server Error'),
};

export const mapErrorCode = (error: ErrorType) => {
  if (error) {
    const { code, message } = error;
    return errorMess[code] ? errorMess[code] : message;
  }
};
