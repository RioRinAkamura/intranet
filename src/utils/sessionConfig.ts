import { Api, createApiClient, SocketIO } from '@goldfishcode/template-api-sdk';
import { SessionStorage } from './SessionStorage';

export const sessionStorage = new SessionStorage();
const config = {
  baseUrl: 'https://template-api.dev.goldfishcode.com/v1',
  authSessionKey: 'AUTH_SESSION_KEY',
  session: sessionStorage,
  socketUrl: 'http://template-ws.dev.goldfishcode.com',
};

const client = createApiClient(config);
export const api = new Api(client);
