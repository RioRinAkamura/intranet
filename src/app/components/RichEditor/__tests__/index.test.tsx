import * as React from 'react';
import { render } from '@testing-library/react';

import { RichEditor } from '..';
import { matchMedia } from 'utils/matchMedia';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

matchMedia();

describe('<RichEditor  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(<RichEditor onSubmit={jest.fn()} />);
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
