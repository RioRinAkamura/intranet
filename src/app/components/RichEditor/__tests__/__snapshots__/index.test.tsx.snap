// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichEditor  /> should match snapshot 1`] = `
.c0 .editor {
  position: relative;
  box-sizing: border-box;
  border: 1px solid #ddd;
  cursor: text;
  padding: 48px 16px 16px 16px;
  border-radius: 2px;
  margin-bottom: 2em;
  box-shadow: inset 0px 1px 8px -3px #ababab;
  background: #fefefe;
  width: 500px;
}

.c0 .editor .public-DraftEditor-content {
  min-height: 140px;
}

.c0 .buttonWrapper {
  display: inline-block;
}

.c0 .button {
  width: 100%;
  background: #fbfbfb;
  color: #888;
  font-size: 18px;
  border: 0;
  vertical-align: bottom;
  height: 34px;
}

.c0 .button svg {
  fill: #888;
}

.c0 .button:hover,
.c0 .button:focus {
  background: #f3f3f3;
  outline: 0;
}

.c0 .active {
  background-color: #ddd;
  color: black;
}

.c0 .active svg {
  fill: black;
}

.c0 .toolbar {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  border: 1px solid #ddd;
  background: #fbfbfb;
  border-radius: 2px;
  box-shadow: 0px 1px 3px 0px rgb(220 220 220);
  z-index: 2;
  box-sizing: border-box;
}

.c0 .hashtag {
  color: rgb(28,65,167);
  background-color: rgba(28,65,167,0.1);
}

.c0 blockquote {
  margin: 1em;
}

<div
  class="c0"
>
  <div
    class="editor"
  >
    <div
      class="DraftEditor-root"
    >
      <div
        class="DraftEditor-editorContainer"
      >
        <div
          aria-autocomplete="list"
          aria-expanded="false"
          class="notranslate public-DraftEditor-content"
          contenteditable="true"
          role="combobox"
          spellcheck="false"
          style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
        >
          <div
            data-contents="true"
          >
            <div
              class=""
              data-block="true"
              data-editor="99e7u"
              data-offset-key="osmp-0-0"
            >
              <div
                class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                data-offset-key="osmp-0-0"
              >
                <span
                  data-offset-key="osmp-0-0"
                >
                  <br
                    data-text="true"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="toolbar"
    >
      <div
        class="buttonWrapper"
      >
        <button
          class="button"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.6 10.79c.97-.67 1.65-1.77 1.65-2.79 0-2.26-1.75-4-4-4H7v14h7.04c2.09 0 3.71-1.7 3.71-3.79 0-1.52-.86-2.82-2.15-3.42zM10 6.5h3c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5h-3v-3zm3.5 9H10v-3h3.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </button>
      </div>
      <div
        class="buttonWrapper"
      >
        <button
          class="button"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M10 4v3h2.21l-3.42 8H6v3h8v-3h-2.21l3.42-8H18V4z"
            />
          </svg>
        </button>
      </div>
      <div
        class="buttonWrapper"
      >
        <button
          class="button"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 17c3.31 0 6-2.69 6-6V3h-2.5v8c0 1.93-1.57 3.5-3.5 3.5S8.5 12.93 8.5 11V3H6v8c0 3.31 2.69 6 6 6zm-7 2v2h14v-2H5z"
            />
          </svg>
        </button>
      </div>
      <div
        class="s6m29i4"
      />
      <div
        class="buttonWrapper"
      >
        <button
          class="button"
          type="button"
        >
          H1
        </button>
      </div>
      <div
        class="buttonWrapper"
      >
        <button
          class="button"
          type="button"
        >
          H2
        </button>
      </div>
      <div
        class="buttonWrapper"
      >
        <button
          class="button"
          type="button"
        >
          H3
        </button>
      </div>
      <div
        class="s6m29i4"
      />
      <div
        class="buttonWrapper"
      >
        <button
          class="button"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 10.5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5zm0-6c-.83 0-1.5.67-1.5 1.5S3.17 7.5 4 7.5 5.5 6.83 5.5 6 4.83 4.5 4 4.5zm0 12c-.83 0-1.5.68-1.5 1.5s.68 1.5 1.5 1.5 1.5-.68 1.5-1.5-.67-1.5-1.5-1.5zM7 19h14v-2H7v2zm0-6h14v-2H7v2zm0-8v2h14V5H7z"
            />
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
          </svg>
        </button>
      </div>
      <div
        class="buttonWrapper"
      >
        <button
          class="button"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 17h2v.5H3v1h1v.5H2v1h3v-4H2v1zm1-9h1V4H2v1h1v3zm-1 3h1.8L2 13.1v.9h3v-1H3.2L5 10.9V10H2v1zm5-6v2h14V5H7zm0 14h14v-2H7v2zm0-6h14v-2H7v2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </button>
      </div>
      <div
        class="buttonWrapper"
      >
        <button
          class="button"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="tukdd6b"
      style="transform: translate(-50%) scale(0); visibility: hidden;"
    >
      <div
        class="bpsgbes"
      >
        <button
          class="b181v2oy"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.6 10.79c.97-.67 1.65-1.77 1.65-2.79 0-2.26-1.75-4-4-4H7v14h7.04c2.09 0 3.71-1.7 3.71-3.79 0-1.52-.86-2.82-2.15-3.42zM10 6.5h3c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5h-3v-3zm3.5 9H10v-3h3.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </button>
      </div>
      <div
        class="bpsgbes"
      >
        <button
          class="b181v2oy"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M10 4v3h2.21l-3.42 8H6v3h8v-3h-2.21l3.42-8H18V4z"
            />
          </svg>
        </button>
      </div>
      <div
        class="bpsgbes"
      >
        <button
          class="b181v2oy"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 17c3.31 0 6-2.69 6-6V3h-2.5v8c0 1.93-1.57 3.5-3.5 3.5S8.5 12.93 8.5 11V3H6v8c0 3.31 2.69 6 6 6zm-7 2v2h14v-2H5z"
            />
          </svg>
        </button>
      </div>
      <div
        class="bpsgbes"
      >
        <button
          class="b181v2oy"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <button
    class="ant-btn ant-btn-primary ant-btn-lg"
    type="button"
  >
    <span>
      Save
    </span>
  </button>
</div>
`;
